### ISSUE TITLE
Review the logic of DefaultStorageClass and useDynamicProvisioning

### ISSUE THREAD

**Issue by daisy-ycguo** [apache/incubator-openwhisk-deploy-kube#376]
While I tried to deploy OpenWhisk with Docker in Mac, I found the dynamic provisioning didn't work by default. In [k8s doc](https://kubernetes.io/docs/concepts/storage/dynamic-provisioning/), it is said "When a default StorageClass exists in a cluster and a user creates a PersistentVolumeClaim with storageClassName **unspecified**, the DefaultStorageClass admission controller automatically adds the storageClassName field pointing to the default storage class."

It looks like leaving `storageClassName` as blank is not **unspecified**. So I rewrite the logic of DefaultStorageClass and useDynamicProvisioning, change the pvc specification to:
```
spec:
{{- if **not** .Values.k8s.persistence.useDynamicProvisioning }}
  storageClassName: {{ .Values.k8s.persistence.defaultStorageClass }}
{{- end }}
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: {{ .Values.db.persistence.size }}
```
It works well with Docker in Mac environment.

The new logic is:
1. If dynamic provisioning is supported by the k8s cluster ( checking if there is default StorageClass by `kubectl get StorageClass` ), set `.Values.k8s.persistence.useDynamicProvisioning` to true. Without any additional configuration, the dynamic provisioning works for every pv claim.
2. If dynamic provisioning is **not** supported by the k8s cluster, set `.Values.k8s.persistence.useDynamicProvisioning` to false and set the value of `.Values.k8s.persistence.defaultStorageClass`. Manually create 5 and more pv, and then deploy OpenWhisk.

I think we should test the new logic in another k8s cluster.

**Comment by daisy-ycguo** [comment]
I tested with IKS.
I failed to figure out a way to use dynamic provisioning, even the `.Values.k8s.persistence.defaultStorageClass` is set to "default", following what the document [IBM IKS for OpenWhisk](https://github.com/apache/incubator-openwhisk-deploy-kube/blob/master/docs/k8s-ibm-public.md#configuring-openwhisk) said:
```
IKS does not provide a properly configured DefaultStorageClass, instead you need to tell the Helm chart to use the default StorageClassName as shown above.
```

It looks like IKS use a wired way to configure dynamic provisioning. I will spend more time on it.
